packages/merge-styles/src/Stylesheet.ts

```diff
@@ -136,35 +228,68 @@ export class Stylesheet {
   /**
    * Gets the singleton instance.
    */
-  public static getInstance(): Stylesheet {
-    _stylesheet = _global[STYLESHEET_SETTING] as Stylesheet;
+  public static getInstance(stylesheetKey?: string): Stylesheet {
+    if (stylesheetKey) {
+      _stylesheet = _global[ADOPTED_STYLESHEETS]?.get(stylesheetKey);
+    } else {
+      _stylesheet = _global[STYLESHEET_SETTING] as Stylesheet;
+    }
 
     if (!_stylesheet || (_stylesheet._lastStyleElement && _stylesheet._lastStyleElement.ownerDocument !== document)) {
       const fabricConfig = _global?.FabricConfig || {};
 
-      const stylesheet = new Stylesheet(fabricConfig.mergeStyles, fabricConfig.serializedStylesheet);
+      const stylesheet = new Stylesheet(fabricConfig.mergeStyles, fabricConfig.serializedStylesheet, stylesheetKey);
       _stylesheet = stylesheet;
-      _global[STYLESHEET_SETTING] = stylesheet;
+      if (stylesheetKey) {
+        if (!_global[ADOPTED_STYLESHEETS]) {
+          _global[ADOPTED_STYLESHEETS] = new EventMap();
+        }
+        _global[ADOPTED_STYLESHEETS]!.set(stylesheetKey, stylesheet);
+        const css = _stylesheet._getConstructibleStylesheet();
+        // css.__yo__ = stylesheetKey;
+        requestAnimationFrame(() => {
+          _global[ADOPTED_STYLESHEETS]!.raise('add-sheet', { key: stylesheetKey, sheet: stylesheet });
+        });
+      } else {
+        _global[STYLESHEET_SETTING] = stylesheet;
+      }
     }
 
     return _stylesheet;
   }
 
-  constructor(config?: IStyleSheetConfig, serializedStylesheet?: ISerializedStylesheet) {
+  constructor(config?: IStyleSheetConfig, serializedStylesheet?: ISerializedStylesheet, stylesheetKey?: string) {
+    // If there is no document we won't have an element to inject into.
+    // const defaultInjectionMode = typeof document === 'undefined' ? InjectionMode.none : InjectionMode.insertNode;
+    const defaultInjectionMode = InjectionMode.unstable_constructibleStylesheet;
     this._config = {
-      // If there is no document we won't have an element to inject into.
-      injectionMode: typeof document === 'undefined' ? InjectionMode.none : InjectionMode.insertNode,
+      injectionMode: defaultInjectionMode,
       defaultPrefix: 'css',
       namespace: undefined,
       cspSettings: undefined,
       ...config,
     };
 
+    if (
+      !SUPPORTS_CONSTRUCTIBLE_STYLESHEETS &&
+      this._config.injectionMode === InjectionMode.unstable_constructibleStylesheet
+    ) {
+      this._config.injectionMode = defaultInjectionMode;
```

Comment:
This will set it back to constructible stylesheet injection mode again, no?

Reply:
Hahah yep :) I just changed `defaultInjectionMode` while I'm doing the early work because it makes testing easier. This will get updated so it works correctly.